// vue-modality-v3/dist/vue-modality-v3.esm.js
import { reactive, computed, onMounted, onBeforeUnmount, toRefs, openBlock, createBlock, Transition, withCtx, createElementBlock, normalizeClass, withModifiers, withKeys, createElementVNode, normalizeStyle, toDisplayString, createCommentVNode, renderSlot } from 'vue';

var script = {
  props: {
    width: {
      type: String,
      default: '400px'
    },
    centered: {
      type: Boolean,
      default: false
    },
    overlay: {
      type: Boolean,
      default: true
    },
    'text-center': {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      default: 'Modal'
    },
    'title-class': {
      type: String,
      default: ''
    },
    'hide-header': {
      type: Boolean,
      default: false
    },
    'hide-footer': {
      type: Boolean,
      default: false
    },
    'ok-title': {
      type: String,
      default: 'Ok'
    },
    'ok-disabled': {
      type: Boolean,
      default: false
    },
    'ok-class': {
      type: String,
      default: ''
    },
    'hide-ok': {
      type: Boolean,
      default: false
    },
    'ok-loading': {
      type: Boolean,
      default: false
    },
    'cancel-title': {
      type: String,
      default: 'Cancel'
    },
    'cancel-disabled': {
      type: Boolean,
      default: false
    },
    'cancel-class': {
      type: String,
      default: ''
    },
    'hide-cancel': {
      type: Boolean,
      default: false
    },
    'no-close-on-backdrop': {
      type: Boolean,
      default: false
    },
    'no-close-on-esc': {
      type: Boolean,
      default: false
    },
    error: {
      type: Boolean,
      default: false
    },
    success: {
      type: Boolean,
      default: false
    }
  },

  setup(props, _ref) {
    let {
      emit
    } = _ref;
    const data = reactive({
      opened: false,
      btnLoading: computed(() => {
        return props.okLoading ? 'vm-btn-loading' : '';
      })
    });
    onMounted(() => {
      window.addEventListener('keyup', e => esc(e));
    });
    onBeforeUnmount(() => {
      window.removeEventListener('keyup', e => esc(e));
    });

    const open = () => {
      emit('open');
      data.opened = true;
    };

    const hide = () => {
      emit('hide');
      data.opened = false;
    };

    const backdrop = () => {
      if (!props.noCloseOnBackdrop) hide();
    };

    const esc = e => {
      if (!props.noCloseOnEsc && e.which === 27) hide();
    };

    const ok = e => {
      if (props.okLoading) return;else emit('ok', e);
    };

    const cancel = e => {
      emit('cancel', e);
    };

    return { ...toRefs(data),
      open,
      hide,
      backdrop,
      esc,
      ok,
      cancel
    };
  }

};

const _hoisted_1 = {
  key: 0
};
const _hoisted_2 = {
  key: 0,
  class: "vm-icon vm-error-icon"
};

const _hoisted_3 = /*#__PURE__*/createElementVNode("div", {
  class: "vm-inner-error-icon"
}, null, -1);

const _hoisted_4 = [_hoisted_3];
const _hoisted_5 = {
  key: 1,
  class: "vm-icon vm-success-icon"
};

const _hoisted_6 = /*#__PURE__*/createElementVNode("span", {
  class: "vm-inner-success-icon"
}, [/*#__PURE__*/createElementVNode("div", {
  class: "vm-success-stem"
}), /*#__PURE__*/createElementVNode("div", {
  class: "vm-success-kick"
})], -1);

const _hoisted_7 = [_hoisted_6];
const _hoisted_8 = {
  key: 1,
  class: "vm-footer"
};
const _hoisted_9 = ["disabled"];
const _hoisted_10 = ["disabled"];
const _hoisted_11 = {
  key: 0
};
const _hoisted_12 = {
  key: 1,
  class: "vm-loader"
};
function render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createBlock(Transition, {
    name: "modal-fade"
  }, {
    default: withCtx(() => [_ctx.opened ? (openBlock(), createElementBlock("div", {
      key: 0,
      tabindex: "-1",
      class: normalizeClass(["vue-modality", {
        'vm-centered': $props.centered,
        'vm-overlay': $props.overlay
      }]),
      onClick: _cache[3] || (_cache[3] = withModifiers(function () {
        return $setup.backdrop && $setup.backdrop(...arguments);
      }, ["self"])),
      onKeyup: _cache[4] || (_cache[4] = withKeys(function () {
        return $setup.esc && $setup.esc(...arguments);
      }, ["esc"]))
    }, [createElementVNode("div", {
      class: normalizeClass(["vue-modality-dialog", {
        'margintop': !$props.centered
      }]),
      style: normalizeStyle(`width: ${$props.width}`)
    }, [!_ctx.hideHeader ? (openBlock(), createElementBlock("header", _hoisted_1, [createElementVNode("span", {
      class: "vm-close-btn",
      onClick: _cache[0] || (_cache[0] = function () {
        return $setup.hide && $setup.hide(...arguments);
      })
    }), createElementVNode("div", {
      class: normalizeClass(["vm-header", {
        'vm-header-border': $props.title.length > 0
      }])
    }, [createElementVNode("span", {
      class: normalizeClass(_ctx.titleClass)
    }, toDisplayString($props.title), 3)], 2)])) : createCommentVNode("", true), createElementVNode("main", {
      class: normalizeClass(["vm-body", {
        'vm-text-center': $props.error || $props.success || _ctx.textCenter
      }])
    }, [$props.error ? (openBlock(), createElementBlock("div", _hoisted_2, _hoisted_4)) : createCommentVNode("", true), $props.success ? (openBlock(), createElementBlock("div", _hoisted_5, _hoisted_7)) : createCommentVNode("", true), renderSlot(_ctx.$slots, "default")], 2), !_ctx.hideFooter ? (openBlock(), createElementBlock("footer", _hoisted_8, [!_ctx.hideCancel ? (openBlock(), createElementBlock("button", {
      key: 0,
      class: normalizeClass(["vm-btn vm-cancel-btn", _ctx.cancelClass]),
      onClick: _cache[1] || (_cache[1] = withModifiers(function () {
        return $setup.cancel && $setup.cancel(...arguments);
      }, ["stop"])),
      disabled: _ctx.cancelDisabled || _ctx.okLoading
    }, toDisplayString(_ctx.cancelTitle), 11, _hoisted_9)) : createCommentVNode("", true), !_ctx.hideOk ? (openBlock(), createElementBlock("button", {
      key: 1,
      class: normalizeClass(["vm-btn vm-ok-btn", [_ctx.okClass, _ctx.btnLoading]]),
      onClick: _cache[2] || (_cache[2] = withModifiers(function () {
        return $setup.ok && $setup.ok(...arguments);
      }, ["stop"])),
      disabled: _ctx.okDisabled
    }, [!_ctx.okLoading ? (openBlock(), createElementBlock("span", _hoisted_11, toDisplayString(_ctx.okTitle), 1)) : createCommentVNode("", true), _ctx.okLoading ? (openBlock(), createElementBlock("div", _hoisted_12)) : createCommentVNode("", true)], 10, _hoisted_10)) : createCommentVNode("", true)])) : createCommentVNode("", true)], 6)], 34)) : createCommentVNode("", true)]),
    _: 3
  });
}

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = "\n@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');\n.vue-modality {\n    font-family: \"Open Sans\", sans-serif;\n    color: #555;\n    font-size: 14px;\n    overflow: hidden;\n    line-height: 1.42857143;\n    position: fixed;\n    height: 100%;\n    width: 100%;\n    overflow: auto;\n    top: 0;\n    left: 0;\n    display: flex;\n    z-index: 9999;\n    justify-content: center;\n}\n@media (max-width: 660px) {\n.vue-modality-dialog {\n        width: 96%!important;\n        height: 98%!important\n}\n}\n.vue-modality-dialog {\n    position: absolute;\n    min-height: 100px;\n    max-height: 100vh;\n    overflow: auto;\n    background: white;\n    border-radius: 3px;\n    box-shadow: 0 1px 3px rgba(0, 0, 0, .2);\n    padding: 12px 15px;\n    display: flex;\n    flex-direction: column;\n}\n.vm-overlay{\n    background: rgba(0, 0, 0, .3);\n}\n.vm-centered {\n    align-items: center;\n}\n.vm-start {\n    padding-top: 10px;\n    align-items: flex-start;\n}\n.vm-close-btn {\n    position: absolute;\n    cursor: pointer;\n    top: 16px;\n    right: 16px;\n    width: 16px;\n    height: 16px;\n    z-index: 9999;\n    opacity: .6;\n}\n.vm-close-btn:hover {\n    opacity: 1;\n}\n.vm-close-btn:before, .vm-close-btn:after {\n    position: absolute;\n    content: '';\n    height: 2px;\n    width: 100%;\n    top: 50%;\n    left: 0;\n    margin-top: -1px;\n    background: rgb(83, 83, 83);\n    border-radius: 100%;\n}\n.vm-close-btn:before {\n    transform: rotate(45deg);\n}\n.vm-close-btn:after {\n    transform: rotate(-45deg);\n}\n.vm-btn-loading {\n    opacity: .6;\n    cursor: default!important;\n}\n.vm-header {\n    font-weight: bold;\n    font-size: 16px;\n    padding-bottom: 10px;\n}\n.vm-header-border {\n    border-bottom: 1px solid #e9e9e9;\n}\n.vm-body {\n    padding: 16px 0;\n    flex: 1;\n    /* flex-direction: column;\n    display: flex; */\n}\n\n/* .vm-body-content-center {\n    justify-content: center;\n} */\n.vm-footer {\n    border-top: 1px solid #e9e9e9;\n    padding: 13px 0 0 0;\n    display: flex;\n    justify-content: flex-end;\n}\n.vm-btn {\n    padding: .3rem .75rem;\n    border-radius: 2px;\n    text-align: center;\n    text-decoration: none;\n    display: inline-block;\n    margin-left: 6px;\n    border: 0;\n    transition: all 0.4s ease;\n    cursor: pointer;\n    font: inherit!important;\n}\n.vm-btn:hover {\n    opacity: .6;\n}\n.vm-btn:disabled {\n    background: #c1c1c1;\n    cursor:default;\n}\n.vm-btn:disabled:hover {\n    opacity: 1;\n}\n.vm-ok-btn {\n    margin-left: 6px;\n    border: 0;\n    background-color: #2a6ec3;;\n    color: white;\n}\n.vm-cancel-btn {\n    background-color: grey;\n    color: white;\n}\n.vm-text-center {\n    text-align: center;\n}\n.margintop {\n    margin-top: 10px;\n}\n.modal-fade-enter-from,\n.modal-fade-leave-active {\n    opacity: 0;\n}\n.modal-fade-enter-active,\n.modal-fade-leave-active {\n    transition: opacity .5s ease\n}\n.vm-icon {\n    width: 60px;\n    height: 60px;\n    border-radius: 50%;\n    background-color: #fff;\n    position: relative;\n    overflow: hidden;\n    display: block;\n    margin: 0 auto;\n    box-sizing: border-box;\n}\n.vm-error-icon {\n    border: 4px solid #e06565;\n    margin-bottom: 12px;\n    color: #e06565;\n}\n.vm-inner-error-icon:after {\n    position: relative;\n    display: inline-block;\n    content: \"\\D7\";\n    left: 1px;\n    bottom: 24px;\n    font-size: 68px;\n    animation-name: jump;\n    animation-duration: .9s;\n}\n@keyframes jump {\n0%   {\n        bottom: 24px;\n}\n40%  {\n        bottom: 36px;\n}\n60% {\n         bottom: 24px;\n}\n80% {\n        bottom: 30px;\n}\n100% {\n         bottom: 24px;\n}\n}\n.vm-success-icon {\n    border: 4px solid #29af18;\n    margin-bottom: 12px;\n}\n.vm-inner-success-icon {\n    display:inline-block;\n    width: 40px;\n    height:40px;\n    transform: rotate(45deg);\n}\n.vm-success-stem {\n    position: absolute;\n    width: 4px;\n    height: 32px;\n    background-color: #29af18;\n    left: 30px;\n    top: 7px;\n    border-radius: 2px;\n    animation-name: stem;\n    animation-duration: .4s;\n}\n@keyframes stem {\n0%   {\n        height: 0px;\n        top: 32px;\n}\n49% {\n        height: 0px;\n        top: 32px;\n}\n100% {\n        height: 32px;\n        top: 7px;\n}\n}\n.vm-success-kick {\n    position: absolute;\n    width: 18px;\n    height: 4px;\n    background-color: #29af18;\n    left: 15px;\n    top: 36px;\n    border-radius: 2px;\n    animation-name: kick;\n    animation-duration: 1s;\n}\n@keyframes kick {\n0%   {\n        width: 3px;\n}\n48% {\n        width: 18px;\n}\n100% {\n        width: 18px;\n}\n}\n.vm-loader {\n    border: 2px solid transparent;\n    border-top: 2px solid #f9f9f9;\n    border-right: 2px solid #f9f9f9;\n    border-bottom: 2px solid #f9f9f9;\n    border-radius: 50%;\n    width: 14px;\n    height: 14px;\n    animation: spin 1s linear infinite;\n}\n@keyframes spin {\n0% { transform: rotate(0deg);\n}\n100% { transform: rotate(360deg);\n}\n}\n";
styleInject(css_248z);

script.render = render;

// Import vue component
// IIFE injects install function into component, allowing component
// to be registered via Vue.use() as well as Vue.component(),

var entry_esm = /*#__PURE__*/(() => {
  // Get component instance
  const installable = script; // Attach install function executed by Vue.use()

  installable.install = app => {
    app.component('VueModalityV3', installable);
  };

  return installable;
})(); // It's possible to expose named exports when writing components that can
// also be used as directives, etc. - eg. import { RollupDemoDirective } from 'rollup-demo';
// export const RollupDemoDirective = directive;

export { entry_esm as default };
